pandas/core/dtypes/dtypes.py

```diff
@@ -429,13 +429,19 @@ def __eq__(self, other: Any) -> bool:
     def __repr__(self) -> str_type:
         if self.categories is None:
             data = "None"
+            dtype = "None"
```

Comment:
Do we have a test that hits this branch?

Reply:
Yes, can't recall which one exactly but got a failure before I added this